(DOCSP-20565) v10 partitioners (#85)
mongodb/docs-spark-connector: source/configuration/read.txt

       - ``SamplePartitioner``
             **Requires MongoDB 3.2**. A general purpose partitioner for
             all deployments. Uses the average document size and random
             sampling of the collection to determine suitable
             partitions for the collection. For configuration 
             settings for the SamplePartitioner, see
             :ref:`conf-samplepartitioner`.

       - ``ShardedPartitioner``
             A partitioner for sharded clusters. Partitions the
             collection based on the data chunks. Requires read access
             to the ``config`` database. For configuration settings for
             the ShardedPartitioner, see :ref:`conf-shardedpartitioner`.

       - ``PaginateBySizePartitioner``
             A slow, general purpose partitioner for all deployments.
             Creates partitions based on data size. Requires a query
             for every partition. For configuration settings for the
             PaginateBySizePartitioner, see 
             :ref:`conf-paginatebysizepartitioner`.

       - ``PaginateIntoPartitionsPartitioner``
             A partitioner that creates a specified number of 

>              partitions for a collection. Uses the count and the 

Reviewer: Avoid mixing a high level description "average document size" with a function name (e.g. "the count [of a collection]") as objects in a conjunction. Instead, consider replacing "the count" with "the total number of documents".

             average document size of a collection to calculate the 

>              number of documents to include in each partition. For 
>              configuration settings for the 
>              PaginateIntoPartitionsPartitioner, see
>              :ref:`conf-paginateintopartitionspartitioner`.

Reviewer: Suggestion: although other entries use the form you used here, in the style guide, the preferred way to introduce additional reading is "To learn more about..."
This can help avoid repeating prepositions ("For X for Y, see...")

```suggestion
             number of documents to include in each partition. To learn more about the ``PaginateIntoPartitionsPartitioner``, see the :ref:`configuration settings description <conf-paginateintopartitionspartitioner>`.
```

Author: Sounds good, applied to all partitioner descriptions

       The connector provides the following partitioners:

       - ``SamplePartitioner``

>              **Requires MongoDB 3.2**. A general purpose partitioner for

Reviewer 2: I think we can remove the Requires MongoDB 3.2 prefix now :)

             to the ``config`` database. To learn more about the 
             ``ShardedPartitioner``, see the 
             :ref:`configuration settings description <conf-shardedpartitioner>`.

       - ``PaginateBySizePartitioner``
             A slow, general purpose partitioner for all deployments.
             Creates partitions based on data size. Requires a query
             for every partition. To learn more about the 
             ``PaginateBySizePartitioner``, see the 
             :ref:`configuration settings description <conf-paginatebysizepartitioner>`.

       - ``PaginateIntoPartitionsPartitioner``
             A partitioner that creates a specified number of 
             partitions for a collection. Uses the total number of 
             documents and the average document size in a collection to 
             calculate the number of documents to include in each 
             partition. To learn more about the 
             ``PaginateIntoPartitionsPartitioner``, see the 
             :ref:`configuration settings description <conf-paginateintopartitionspartitioner>`.

       You can also specify a custom partitioner implementation. For

>        custom implementations of the ``Partitioner`` trait, provide

Reviewer 2: -trait + interface

Its now in Java so no traits.

>        the full class name. If you don't provide package names, then

Reviewer 2: Remove the `If you don't provide package names` sentence - this is no longer the case.

Partitioner Configuration
~~~~~~~~~~~~~~~~~~~~~~~~~

> .. _conf-samplepartitioner:

Reviewer: I didn't check whether the old ref was linked to by anything, or whether there might be a search engine pointing to this section, but it's safest to keep the old ref as well as create the new one.

E.g.
```
.. _conf-mongosamplepartition:
.. _conf-samplepartitioner:
```

     - Description

>    * - ``partitionKey``

Reviewer 2: All partitioner configuration has the following prefix: `partitioner.options.` so it should be `partitioner.options.partitionKey`

.. example::

   For a collection with 640 documents with an average document 
   size of 0.5 MB, the default ``SamplePartitioner`` configuration 
   values creates 5 partitions with 128 documents per partition.

   The MongoDB Spark Connector samples 50 documents (the default 10 
   per intended partition) and defines 5 partitions by selecting 
   ``partitionKey`` ranges from the sampled documents.

.. _conf-mongoshardedpartitioner:
.. _conf-shardedpartitioner:

> ``ShardedPartitioner`` Configuration

Reviewer 2: The shardKey option has been removed - we determine this automatically.

   * - Property name
     - Description

   * - ``partitioner.options.partitionKey``
     - A comma-delimited list of fields by which to split the 
       collection data. The fields should be indexed and contain unique 
       values.

       **Default:** ``_id``

>    * - ``maxNumberOfPartitions``

Reviewer 2: Just missed the prefix here!
